### PR TITLE
Reduce time between editing contacts and seeing those changes in the app

### PR DESCRIPTION
**Pull request by michaelkirk**
* Move a couple lib methods behind our own interface
* Cache parsing phone numbers since it's expensive
* I considered caching formatting as well since it's also a bit
  expensive, but generating an appropriate cache key was actually slower
  than the raw implementation.

PTAL @charlesmchen

### CODE REVIEW

File: src/Contacts/PhoneNumberUtil.m [hunk 1]
```diff
@@ -32,13 +34,45 @@ - (instancetype)init {
...
+                    defaultRegion:(NSString *)defaultRegion
+                            error:(NSError **)error
+{
+    NSString *hashKey = [NSString stringWithFormat:@"numberToParse:%@defaultRegion:%@", numberToParse, defaultRegion];
+
+    NBPhoneNumber *result = [self.parsedPhoneNumberCache objectForKey:hashKey];
+
+    if (!result) {
+        result = [self.nbPhoneNumberUtil parse:numberToParse defaultRegion:defaultRegion error:error];
+        if (result) {
```
Comment (charlesmchen) [inline src/Contacts/PhoneNumberUtil.m]: `if (result && !error) {`

and log error if non-nil?

Comment (charlesmchen) [reply src/Contacts/PhoneNumberUtil.m]: Also `error` parameter will be nil the second time we try to parse an invalid phone number, which might affect logging at caller. NBD?

**Review by charlesmchen — APPROVED** [review]
LGTM

File: src/Contacts/PhoneNumberUtil.m (hunk 1 above)
Comment (michaelkirk) [reply src/Contacts/PhoneNumberUtil.m]: Good call, I opted to just never cache when there's an error.

**Pull request merged**